plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -160,6 +160,54 @@ public void testDeleteRowsConcurrently()
         }
     }
 
+    private long getLatestSnapshotId(String tableName)
+    {
+        return (long) computeScalar(format("SELECT snapshot_id FROM \"%s$snapshots\" ORDER BY committed_at DESC FETCH FIRST 1 ROW WITH TIES", tableName));
+    }
+
+    @Test
+    public void testCreateOrReplaceTable()
+    {
```

Comment:
```
trino> create table iceberg.default.table1(c1 integer, c2 integer)  with (format='PARQUET' , sorted_by = ARRAY['c2']);
CREATE TABLE
trino> insert into iceberg.default.table1 values (1,10) , (2,20);
INSERT: 2 rows

trino> show create table iceberg.default.table1;
                                              Create Table                                              
--------------------------------------------------------------------------------------------------------
 CREATE TABLE iceberg.default.table1 (                                                                  
    c1 integer,                                                                                         
    c2 integer                                                                                          
 )                                                                                                      
 WITH (                                                                                                 
    format = 'PARQUET',                                                                                 
    format_version = 2,                                                                                 
    location = 'hdfs://hadoop-master:9000/user/hive/warehouse/table1-85b38144b3d843778f141f9bbe964bf1', 
    sorted_by = ARRAY['c2 ASC NULLS FIRST']                                                             
 )                                                                                                      
(1 row)

trino> create or replace table iceberg.default.test1 as select 1 c1, 10 c2;
CREATE TABLE: 1 row

trino> select * from iceberg.default.test1;
 c1 | c2 
----+----
  1 | 10 
(1 row)


trino> show create table iceberg.default.table1;
                                              Create Table                                              
--------------------------------------------------------------------------------------------------------
 CREATE TABLE iceberg.default.table1 (                                                                  
    c1 integer,                                                                                         
    c2 integer                                                                                          
 )                                                                                                      
 WITH (                                                                                                 
    format = 'PARQUET',                                                                                 
    format_version = 2,                                                                                 
    location = 'hdfs://hadoop-master:9000/user/hive/warehouse/table1-85b38144b3d843778f141f9bbe964bf1', 
    sorted_by = ARRAY['c2 ASC NULLS FIRST']                                                             
 )                                                                                                      
(1 row)
```

I would be expecting that the "old" properties of the tables are not present anymore and are replaced by defaults in case that CTAS statement is not specifying them explicitly.
Note that `sorted_by` is still present in the table properties, even though the table has been replaced.

Reply:
The table comment is also maintained after `CREATE OR REPLACE`.

What are the semantics of `CREATE OR REPLACE` statement?

Reply:
I have given this a bit more thought and I think the Spark table properties merge behaviour is not desired. 

It puts unnecessary complexity onto the user. It breaks the idempotency of the operation as the output depends on the earlier state, in other words executing the same query yields different results based on the history of the table. I think that is not desired.

For example,

```
CREATE TABLE a WITH (partitioning=ARRAY['a']) AS SELECT 1 a;
CREATE OR REPLACE TABLE a AS SELECT 1 b;
```

The resulting table would have partitioning with an invalid column, just by looking at this single statement it would not be clear that this is a partitioned table. I think that is not desired, for example, this last CREATE table statement could be in source code and if they execute only the last statement against a new database, the result would be different, ie idempotency is broken.

Reply:
I added some more tests around this behaviour, adding/removing table properties, table and column comments.